### PR TITLE
Adding basic Keycloak operator

### PR DESCRIPTION
**Pull request by almahmoud**
Codifying the new keycloak operator in the playbook
Friends with https://github.com/CloudVE/projman-helm/pull/7 https://github.com/CloudVE/cloudman-helm/pull/76 https://github.com/galaxyproject/cloudman/pull/146

### CODE REVIEW

File: roles/cloudman-boot/templates/cm_chart_values.yml.j2 [hunk 1]
```diff
@@ -19,26 +19,26 @@ cloudlaunch:
...
-                root_url: "https://{{ cluster_hostname }}/grafana"
-             auth.generic_oauth:
-                auth_url: "https://{{ cluster_hostname }}/auth/realms/master/protocol/openid-connect/auth"
-                token_url: "https://{{ cluster_hostname }}/auth/realms/master/protocol/openid-connect/token"
-                api_url: "https://{{ cluster_hostname }}/auth/realms/master/protocol/openid-connect/userinfo"
+influxdb:
+    setDefaultUser:
+        user:
+            password: {{ random_alphanumeric(16) }}
+kube-prometheus-stack:
```
Comment (nuwang) [inline roles/cloudman-boot/templates/cm_chart_values.yml.j2:26]: This should be kubePrometheusStack I believe, or the dash trips things up?

Comment (almahmoud) [reply roles/cloudman-boot/templates/cm_chart_values.yml.j2]: I was trying with the version that is currently on master following: https://github.com/CloudVE/cloudman-helm/pull/75
The camelcase is not ideal cause it then gets used for the release name and it's not DNS compliant. I've defaulted to `kubeprometheus` as an alias instead for now.

File: roles/cloudman-boot/tasks/auth.yaml [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+- name: Apply the rendered GVL realm
+  command: /usr/local/bin/kubectl apply -f {{ kc_path }}/deploy/gvl-keycloakrealm.yaml -n {{ cm_keycloak_namespace }}
+
+- name: Render Keycloak ingress cert patch
+  template:
+    src: keycloak_ingress_patch.yml.j2
+    dest: "{{ kc_path }}/deploy/cloudman-keycloak-ingress-patch.yaml"
+  tags: auth
+
+- name: Check if Rancher SAML Client Certificate is already created
```
Comment (nuwang) [inline roles/cloudman-boot/tasks/auth.yaml:57]: I think it's worth moving the rancher-auth-integration stuff to a separate file?

**Review by nuwang — COMMENTED** [review]
I think the PR is looking pretty good. One issue is that there are a large number of commits with generic commit messages. It would be good to rebase and clean up the commit history and merge related things together so commits can, as much as possible, be understood as a related unit of change. Have also made some additional minor suggestions, this is looking great overall and I cannot wait to switch to it!

File: roles/cloudman-boot/tasks/auth.yaml (hunk 2 above)
Comment (almahmoud) [reply roles/cloudman-boot/tasks/auth.yaml:57]: I was originally doing that, but it does make it overall slower with not being able to alternate. Right now i'm doing things like generating the rancher SAML client while waiting for the operator to reconcile the instance and create the deployment/ingress. It adds a few minutes if we separate them and wait for keycloak first then install rancher and wait for it which is what I was originally doing. I thought the block nesting would be enough for the visual understanding. I was also thinking maybe it'd make sense to start tagging things more aggressively to make it easier to run only a specific logical grouping of tasks when we want to? But the latter should probably be a separate PR adding tags throughout the playbook

File: roles/cloudman-boot/defaults/main.yml [hunk 3]
```diff
@@ -5,17 +5,29 @@ cm_initial_storage_size: "{{ lookup('env', 'CM_INITIAL_STORAGE_SIZE') }}"
...
+rancher_hostname: "rancher.{{ cluster_hostname }}{% if cluster_hostname | ipaddr %}.xip.io{% endif %}"
+
+# Keycloak operator
+cm_keycloak_operator_version: "{{ lookup('env', 'CM_KEYCLOAK_OPERATOR_VERSION') | default('latest', true) }}"
+cm_keycloak_namespace: "{{ lookup('env', 'CM_KEYCLOAK_NAMESPACE') | default('keycloak', true) }}"
+cm_keycloak_operator_fork: "{{ lookup('env', 'CM_KEYCLOAK_OPERATOR_FORK') | default('keycloak/keycloak-operator', true) }}"
+cm_keycloak_realm_name: "{{ lookup('env', 'CM_KEYCLOAK_REALM_NAME') | default('gvl', true) }}"
+cm_keycloak_realm_registration: "{{ lookup('env', 'CM_KEYCLOAK_REALM_REGISTRATION') | default('false', true) }}"
+keycloak_hostname: "auth.{{ cluster_hostname }}{% if cluster_hostname | ipaddr %}.xip.io{% endif %}"
+
```
[the remark this answers is not in the document]
Comment (almahmoud) [reply roles/cloudman-boot/defaults/main.yml:30]: This seems like a good idea, but I'll rename it to `cm_keycloak_operator_path` to be a bit more in line with the other vars

File: roles/cloudman-boot/tasks/auth.yaml (hunk 2 above)
Comment (nuwang) [reply roles/cloudman-boot/tasks/auth.yaml:57]: Ok sounds good. Only some of it is block nested though?

**Pull request merged**